src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs

```diff
@@ -0,0 +1,220 @@
+                        InstanceId = row.InstanceId,
+                        IsSystemMessage = row.IsSystemMessage,
+                        MessageId = row.MessageId,
+                        MessageIntent = row.MessageIntent,
+                        MessageType = row.MessageType,
+                        ProcessedAt = row.ProcessedAt,
+                        ProcessingTime = row.ProcessingTime,
+                        Status = row.Status,
+                        TimeSent = row.TimeSent
+                        //Headers = row.Headers,
+                        //InvokedSagas = row.InvokedSagas,
+                        //OriginatesFromSaga = row.OriginatesFromSaga,
+                        //ReceivingEndpoint = row.ReceivingEndpoint,
+                        //SendingEndpoint = row.SendingEndpoint
+                    });
+                }
+            }).ConfigureAwait(false);
+
+            return new QueryResult<IList<MessagesView>>(messagesViews, new QueryStatsInfo(string.Empty, messagesViews.Count));
+        }
+
+        public async Task<QueryResult<IList<MessagesView>>> QueryMessagesByReceivingEndpoint(bool includeSystemMessages, string endpointName, PagingInfo pagingInfo, SortInfo sortInfo)
+        {
+            var messagesViews = new List<MessagesView>();
+
+            await connectionManager.Perform(async connection =>
+            {
+                var rows = await connection.QueryAsync(
+                    @"SELECT * FROM [dbo].[MessagesView] WHERE [ReceivingEndpointName] = @receivingEndpointName",
+                    new
+                    {
+                        receivingEndpointName = endpointName
+                    }).ConfigureAwait(false);
+
+                foreach (var row in rows)
+                {
+                    messagesViews.Add(new MessagesView
+                    {
+                        BodySize = row.BodySize,
+                        BodyUrl = row.BodyUrl,
+                        ConversationId = row.ConversationId,
+                        CriticalTime = row.CriticalTime,
+                        DeliveryTime = row.DeliveryTime,
+                        Id = row.Id,
+                        InstanceId = row.InstanceId,
+                        IsSystemMessage = row.IsSystemMessage,
+                        MessageId = row.MessageId,
+                        MessageIntent = row.MessageIntent,
+                        MessageType = row.MessageType,
+                        ProcessedAt = row.ProcessedAt,
+                        ProcessingTime = row.ProcessingTime,
+                        Status = row.Status,
+                        TimeSent = row.TimeSent
+                        //Headers = row.Headers,
+                        //InvokedSagas = row.InvokedSagas,
+                        //OriginatesFromSaga = row.OriginatesFromSaga,
+                        //ReceivingEndpoint = row.ReceivingEndpoint,
+                        //SendingEndpoint = row.SendingEndpoint
```

Comment:
Same as above

Reply:
It's what's left to do so I've added TODO comments above those to make it clear